### ISSUE TITLE
[Request] SCDoc modal tag: post

### ISSUE THREAD

**Issue by prko** [supercollider/supercollider#6304]
## Motivation
There is currently no built-in way to display coloured text as in the Post window when including the Post window return in Help documents. The only way is to use images, which is unproductive. 

## Description of Proposed Feature
The block code shows the colour of the code:
```
code::
SinOsc.ar
::
```
Like this, it would be good to have a `post` tag.

## Plan for Implementation

Unfortunately I do not know C++. If someone gives me instructions I could try it, otherwise we should wait for a developer who feels the same need.

**Comment by telephon** [comment]
The coloring in the post window is currently quite inconsistent. So we need to fix that first. When you post something that has several lines, it will highlight the first line only, because it starts with a "->". I also find it inconvenient that the "->" is appended, because when you copy the code, you have to get rid of it in an extra step.

**Comment by capital-G** [comment]
I really like this idea, and IMO it would be a good way to add an even more general tag `div`, which the current documentation lacks. Due to this lack it makes it difficult/impossible to create truly new designs for the docs as the placement and look of individual blocks can not be accessed or changed in a consistent manner.

If we manage to introduce a kind of div wrapper with a `class` variable, any kind of visual or interactive could be handled in traditional css or js ways as it seems less a concern of code generation from within scdocs but an enhancement of already generated output.

Myst has something like this as well: https://myst-parser.readthedocs.io/en/v0.17.1/syntax/optional.html#syntax-admonitions

**Comment by prko** [comment]
@telephon 
> ... So we need to fix that first. ....

Yes, even though I am used to it, the inconsistency sometimes confuses me. If the order of development is as you say, there is no other way but to wait...

---

@capital-G 
>IMO it would be a good way to add an even more general div tag, which the current documentation lacks.

The easiest way to do this is to enable the necessary HTML tags directly in SCDoc, without having to create SCDoc modal tags, which is done in cooperation between C++ (`SCDoc.y`, `SCDoc.l`, etc) and SCClasslibrary (`SCDoc.sc` and `SCDocRenderer.sc`) as in my recent PR:

[SCDoc improvements: 3. Some HTML tags are enabled](#6265)

However, I think none of the developers would see this as positive, as the SCDoc syntax will be ruined....

My PR #6256 may be based on my amateur experience of programming skills, but it is flexible. Also, considering that sclang is not the only syntax style, the direct use of HTML in SCDoc does not seem so bad.... Also, considering the speed of maintenance and development of SC..., hm...

**Comment by smoge** [comment]
> If we manage to introduce some kind of div wrapper with a `class` variable, any kind of visual or interactive could be handled in a traditional css or js way, as it seems less a matter of generating code from within scdocs, but an enhancement of the already generated output.



A new syntax element that allows custom classes or IDs would require some change in the grammar (bison/flex files), but that could be minimal.

This could look something like "div::" classlist body "::", where classlist is a list of CSS classes or an ID.

```
divwrapper        ::= "DIV::" classlist body "::"
classlist         ::= { anyword COMMA } anyword
```

```
classlist ::= classname { COMMA classname }
classname ::= [a-zA-Z_][a-zA-Z0-9_-]*
```

Maybe something like this?

```
div:: special-style3
This is the content of the div element which will be styled accordingly.
::
```

(sorry if misunderstood)

**Comment by smoge** [comment]
@capital-G  Your idea of an HTML:: that bypasses the renderer could solve some situations. (I think it could be helpful for MathJax, right @prko ?)
